### PR TITLE
Mccalluc/subviewer

### PR DESCRIPTION
**Pull request by mccalluc**
Treat this less as something that actually should be merged, and more as a basis for questions.

Use case: I want use HiGlass literally behind the scene in Vitessce. Right now, the background imagery is just an SVG containing an `image`: As deck.gl in the foreground reports drag events, the viewBox on the svg is updated. -> I was hoping to use HiGlass instead, except with `zoomTo` instead of tweaking svg props.

1) Is there any good reason to ever to use a 0 animation time in zoomTo? I'm changing position 10 times per second: with no animation, it was not keeping up, but even an animation time of 1 looks great... If there's no good reason to use animation time of 0, would we like to make the default 1, instead?

2) Do you think I could implement the whole colorbar as SVG, and not just the sliders? As I would be using it, it would be covered by other layers. If it's not something higlass provides, then I might end up reimplementing it in Vitessce, which seems like a waste.

3) Do you have any interest in adding a `trackViewer` like this to the API? It's trivial for me to implement it on the Vitessce side, but maybe it would make this kind of use easier in the future?

### CODE REVIEW

**Comment by pkerpedjiev** [comment]
> Is there any good reason to ever to use a 0 animation time in zoomTo? I'm changing position 10 times per second: with no animation, it was not keeping up, but even an animation time of 1 looks great... If there's no good reason to use animation time of 0, would we like to make the default 1, instead?

0 should be an immediate transition but it sounds like 0 is actually slower for you than 1? If that's the case, I'm fine with changing it to 1.

> Do you think I could implement the whole colorbar as SVG, and not just the sliders? As I would be using it, it would be covered by other layers. If it's not something higlass provides, then I might end up reimplementing it in Vitessce, which seems like a waste.

I think we may have originally had that. The main reason it's rendered on the canvas instead of SVG is that it's easier to export as PNG and it stays in the same position as the heatmap. If it's rendered as SVG, it'll be rendered above all other canvas-based tracks. Neither solution is ideal since if we render on the canvas layer, other tracks can occlude the colorscale. See this, for example: https://github.com/higlass/higlass/issues/490

Other suggestions for how to handle this are welcome. And we should actually have an SVG implementation in `exportAsSVG`.

> Do you have any interest in adding a trackViewer like this to the API? It's trivial for me to implement it on the Vitessce side, but maybe it would make this kind of use easier in the future?

Yes, definitely. Maybe eventually we can break that out into its own library to keep the size down.

File: app/scripts/hglib.js [hunk 1]
```diff
@@ -102,3 +102,53 @@ export const viewer = (element, viewConfig, options) => {
 
   return hg.api;
 };
+
+export const trackViewer = (element, [xMin, xMax, yMin, yMax], trackConfig, options) => {
+  if (!trackConfig.options.colorbarPosition) {
+    trackConfig.options.colorbarPosition = 'hidden';
+    // If the colorbar were all SVG, I could change the colorbar Z position,
+    // and move the whole thing around, but apart from the handlebars,
+    // I think it's all part of the canvas, right?
```
Comment (pkerpedjiev) [inline app/scripts/hglib.js]: Yes. See here: https://github.com/higlass/higlass/blob/fdd8f08c3769849a129f5f98c7d6ffd8a3106dd8/app/scripts/HeatmapTiledPixiTrack.js#L507

File: app/scripts/hglib.js [hunk 2]
```diff
@@ -102,3 +102,53 @@ export const viewer = (element, viewConfig, options) => {
...
+  }
+  if (!trackConfig.options.labelTextOpacity) {
+    trackConfig.options.labelTextOpacity = 0;
+    // TODO: There is still a rectangle in the lower right... Is there an easy way to just turn
```
Comment (pkerpedjiev) [inline app/scripts/hglib.js]: You should be able to set `labelPosition` to `hidden`. You could also set `labelBackgroundOpacity` to 0.

**Comment by mccalluc** [comment]
@pkerpedjiev : When #626 is merged, I would like to rebase this, see it work locally, and then merge it. Are there any issues here which you would consider blockers?

**Comment by pkerpedjiev** [comment]
#626 is merged. I updated your PR to include a docstring. Could you go over it and see if it makes sense and agrees with your intention? I also removed the unused options parameter.

Why are you returning {id, hgApi} instead of just hgApi? If you're setting id to the same thing all the time, why not just say what it is in the docstring and not return it?

**Comment by mccalluc** [comment]
> Why are you returning {id, hgApi} instead of just hgApi? If you're setting id to the same thing all the time, why not just say what it is in the docstring and not return it?

I think I framed this as a don't-repeat-yourself issue, and as a maintainability issue.
- DRY: It's better to have the literal value in one place than three (where it's used internally, where it's documented, and where it would be used externally).
- Maintainability: If I'm looking at my code in the future, and there's some magic string in it, I may wonder where it came from. Hopefully all three instances of the value will match, but if not...

I'll look at the other changes now.

**Comment by mccalluc** [comment]
@pkerpedjiev : I think this is ready to go now. Regarding the `id` that is returned, I could change it. I like the way it is, but I could be looking at the problem wrong.

I've regenerated `examples.html` to include the example I just added: It had been pretty-printed at somepoint, and I could add that to the script, if it's useful to anyone... but since I've thought of it as something that would just be regenerated periodically, I don't see it as a priority.

**Comment by mccalluc** [comment]
@pkerpedjiev : ping!

**Pull request merged**